sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/ClientRetryPolicy.java

```diff
@@ -152,6 +160,33 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
         return this.throttlingRetry.shouldRetry(e);
     }
 
+      private boolean gatewayRequestCanFailoverOnTimeout(RxDocumentServiceRequest request, CosmosException clientException) {
+        //Query Plan requests
+        if(request.getResourceType() == ResourceType.Document
+            && request.getOperationType() == OperationType.QueryPlan) {
+            return true;
+        }
+
+        //Meta data request check
+        boolean isMetaDataRequest = (request.getOperationType() != OperationType.ExecuteJavaScript
+            && request.getResourceType() == ResourceType.StoredProcedure)
+            || request.getResourceType() != ResourceType.Document;
```

Comment:
This might not be true for all resource types other than Document, like `PartitionKey`, or `Index`, or something else. I wonder if this if else condition has been discussed offline with the broader team. If not, may be we should.

Reply:
This is the same check we have used previously on the [.NET end](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/ff01c1448fe40226d2a8433c135e48d47f66d9b9/Microsoft.Azure.Cosmos/src/Routing/LocationCache.cs#L211) to determine if a request is a metadata operation.